crates/ty_python_semantic/src/types/infer/builder/type_expression.rs

```diff
@@ -148,15 +148,13 @@ impl<'db> TypeInferenceBuilder<'db, '_> {
                     // anything else is an invalid annotation:
                     op => {
                         self.infer_binary_expression(binary, TypeContext::default());
-                        if let Some(mut diag) = self.report_invalid_type_expression(
+                        self.report_invalid_type_expression(
                             expression,
                             format_args!(
                                 "Invalid binary operator `{}` in type annotation",
                                 op.as_str()
                             ),
-                        ) {
-                            diag.info("Did you mean to use `|`?");
-                        }
```

Comment:
we don't have any tests for this subdiagnostic and I'm not convinced it's likely to be helpful. If a user has `int + str` in a type expression, I think it's very unclear what they meant; if they have `int - bool`, I think it's likely that they were trying to write a negation type rather than a union type; etc.